docs/manual/src/internals/rendering_foreign_bindings.md

```diff
@@ -2,84 +2,69 @@
-limitations).
-
-### Adding imports
-
-We want our type template files to specify what needs to be imported, but we don't want it to render the import
-statements directly. The imports should be rendered at the top of the file and de-duped in case multiple types require
-the same import. We handle this by:
-
-  - Defining a separate Askama template struct that loops over all types and renders the definition/helper code for them.
-  - That struct also stores a `BTreeSet` that contains the needed import statements and has an `add_import()` method that
-    the template code calls.  Using a `BTreeSet` ensures the imports stay de-duped and sorted.
-  - Rendering this template as a separate pass.  The rendered string and the list of imports get passed to the main
-    template which arranges for them to be placed in the correct location.
-
-### Including templates once
-
-We want our type template files to render runtime code, but only once.  For example, we only want to render
-`ObjectRuntime.kt` once, even if there are multiple Object types defined in the UDL file.  To handle this the type
-template defines an `include_once_check()` method, which tests if we've included a file before.  The template code then
-uses that to guard the Askama `{% include %}` statement.  See [`Object.kt` for an
-example](https://github.com/mozilla/uniffi-rs/blob/470740289258e1f06171a976d8e15978f028e391/uniffi_bindgen/src/bindings/kotlin/templates/ObjectTemplate.kt#L2)
+The templates have access to `ci`, a [`ComponentInterface`](https://github.com/mozilla/uniffi-rs/blob/main/uniffi_bindgen/src/interface/mod.rs), which is the Rust representation of all the UniFFI types in your crate.
+
+The task of the templates is to render `ci` into a "module" for the foreign binding.  This mainly consists of rendering support for each [`Type`](https://github.com/mozilla/uniffi-rs/blob/main/uniffi_meta/src/types.rs) described in your crate.
+
+Eg, here's where [Python uses `ci` to iterate over the types](https://github.com/mozilla/uniffi-rs/blob/884f7865f3367c494e9165e21c1255018577db01/uniffi_bindgen/src/bindings/python/templates/Types.py#L3)
+
+The templates create foreign-native types for everything from ffi-native types (int/etc) to functions, dictionaries etc. The implementation of these generated types might call your your Rust implemented FFI, as described below.
+
+Bidings also need to do alot of work to make language identifiers etc work correctly - eg, turn `this_func(this_arg: ThisType)` into `thisFunc(...)`
+
+## Breaking down a Rust function called by Python.
+
+Let's take a quick look at where [Python generates a top-level public function](https://github.com/mozilla/uniffi-rs/blob/884f7865f3367c494e9165e21c1255018577db01/uniffi_bindgen/src/bindings/python/templates/TopLevelFunctionTemplate.py#L37-L40+)
+
+It's making, eg `def this_func(this_arg=0) -> None:` - let's break it down:
+
+`def {{ func.name()|fn_name }}({%- call py::arg_list_decl(func) -%}) -> None:`
```

Comment:
Not sure how it's rendered, but might be better as a code block instead of inline:

````suggestion
```
def {{ func.name()|fn_name }}({%- call py::arg_list_decl(func) -%}) -> None:
```
````